packages/sanity/src/presentation/loader/LiveQueries.tsx

```diff
@@ -483,7 +483,11 @@ export function turboChargeResultIfSourceMap<T = unknown>(
       }
       return changedValue
     },
-    perspective,
+    // TODO: Update applySourceDocuments to support releases.
+    Array.isArray(perspective) &&
```

Comment:
huh interesting, why are we doing this?

Reply:
Also jsut realised this code is the same changes as applied in the  [LoaderQueries.tsx](https://github.com/sanity-io/sanity/pull/8317/files#diff-fc19b0caf7b88a8fc299ff32abf4dd08ab8dc56dcb78dfe91f74ea41753aae4c) should it be moved somewhere else so both can use it? Might be overengineering but since it looks to be the same code in somewhat seemingly related spaces

Reply:
the idea is for LiveQueries to replace LoaderQueries relatively soon. The rule of thumb we follow is that since only one of them will live on, we only make refactors that continue to make sense once LoaderQueries are deleted.
Does that make sense?

Reply:
I think so, just wonder if we should write that explicitly on the component so the question doesn't come up later if we need to make more changes to both components 🤔